lib/primitives/mtx.js

```diff
@@ -1688,13 +1711,58 @@ class CoinSelector {
    */
 
   fund() {
-    // Ensure all preferred inputs first.
-    if (this.inputs.size > 0) {
-      const coins = [];
+    this.resolveInputs();
+    this.resolvePreferred();
+
+    if (this.isFull())
+      return;
+
+    while (this.index < this.coins.length) {
+      const coin = this.coins[this.index++];
+
+      if (!this.isSpendable(coin))
+        continue;
+
+      this.tx.addCoin(coin);
+      this.chosen.push(coin);
+
+      if (this.selection === 'all')
+        continue;
 
-      for (let i = 0; i < this.inputs.size; i++)
-        coins.push(null);
+      if (this.isFull())
+        break;
+    }
+  }
+
+  /**
+   * Resolve coins for existing inputs.
+   */
+
+  resolveInputs() {
+    if (this.inputs.size === 0)
+      return;
 
+    // Ensure we have coins for the existing inputs.
+    const view = this.tx.view;
```

Comment:
This is confusing me a little bit because the constructor already assigns `this.tx = tx.clone();` and `this.view = tx.view;` Except for the last `for` loop (line 1782) you use `this.view` which is what I'd expect ...?